src/app/components/Search.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+import { ApplicationLauncher, ApplicationLauncherItem } from '@patternfly/react-core'
+import { useDocumentTitle } from '@app/utils/useDocumentTitle'
+import  fuzzysort from 'fuzzysort'
+import { SearchIcon } from '@patternfly/react-icons'
+import { Link } from 'react-router-dom'
+
+
+const Search: React.FunctionComponent<{ title: string }> = ({ title }) => {
+    const [data, setData] = React.useState([])
+    const [results, setresults] = React.useState([])
+    const [outputFocus, setOutputFocus] = React.useState(false)
+    const [inputFocus, setInputFocus] = React.useState(false)
+
+    useDocumentTitle(title)
+
+    const loadData = () => {
+        fetch(`https://imagedirectory.cloud/images/v1/idx/list/image-names`, {
+          method: 'get',
+        })
+        .then(res => res.json())
+        .then(data => {
+          setData(data)
+        })
+    }
+
+    React.useEffect(() => {
+        loadData()
+    })
```

Comment:
I guess `loadData` should only run when `Search` is getting mounted. Therefore I would recommend to add an empty dependency array, otherwise the it would run on every render.
```suggestion
    React.useEffect(() => {
        loadData()
    }, [])
```
Same for the effect in `src/app/Pages/Browser/Search.tsx`, not sure what will stay :slightly_smiling_face:

Reply:
The behavior was already discovered, but interesting fix :smile:

Reply:
> I guess `loadData` should only run when `Search` is getting mounted. Therefore I would recommend to add an empty dependency array, otherwise it would run on every render.
> 
> Same for the effect in `src/app/Pages/Browser/Search.tsx`, not sure what will stay 🙂

Nice, great find! The sad thing is I ran into an issue on the image detail view in which the component stayed mounted after selecting a new image version which had the exact same fix (by monitoring the props passed in with the path) but I completely ignore this onEffect call.. *face palm*